packages/hdwallet-provider/src/index.ts

```diff
@@ -215,18 +225,94 @@ class HDWalletProvider {
     });
   }
 
-  public send(
+  private initialize(): Promise<void> {
+    return new Promise(resolve => {
+      this.sendAsync({
+        jsonrpc: '2.0',
+        id: Date.now(),
+        method: 'eth_chainId',
+        params: []
+      }, (_error, response) => {
```

Comment:
We should `reject` if there is an error, or if the response doesn't contain a valid QUANTITY in the `result`.

Reply:
I agree, though we don't have a concept of `QUANTITY` in Truffle so maybe just for the error and check that it's a number.

Reply:
As long as it's a hex string it's probably fine.

Teccchnically the result could be greater than `Number.MAX_SAFE_INTEGER`, but I'm not sure this is an edge case worth handling.